From 70cf26ed73e3b58eae6f42433363bd9801114a89 Mon Sep 17 00:00:00 2001
From: Santhana Kumar <santhanakumar.a@ltts.com>
Date: Fri, 5 Apr 2019 19:50:54 +0530
Subject: [PATCH 42/47] Add simple on-module I2C devices.

An NXP PCF8563 RTC is connected to I2C1: The i.MX6-internal SNVS draws about
40 uA, making it unacceptable as coin-cell powered RTC.

(Note: The defconfig compiles the SNVS RTC as module only, which ensures the
PCF8563 becomes primary clock (rtc0). It also disables auto-sync with NTP to
avoid little bursts of I2C traffic, silently, every 11 minutes, in case a
custom baseboard adds something clever to I2C1.)

The ADLINK SEMA Board Management Controller (BMC) is attached to I2C3. It's
just a device tree entry; maybe some day a proper driver will be written,
but currently everything is done in user space.

Signed-off-by: Santhana Kumar <santhanakumar.a@ltts.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index 8d89ae4..ebbbc2a 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -530,6 +530,11 @@
 	pinctrl-0 = <&pinctrl_i2c1>;
 	status = "okay";
 
+	pcf8563@51 {				/* Real Time Clock */
+		compatible = "nxp,pcf8563";
+		reg = <0x51>;
+	};
+
 	pex8605@58 {				/* optional PCIe switch */
 		compatible = "plx,pex8605"; /* guessing, no driver found */
 		reg = <0x58>;
@@ -744,6 +749,11 @@
 	pinctrl-0 = <&pinctrl_i2c3>;
 	status = "okay";
 
+	sema_bmc@28 {			/* SEMA Board Management Controller */
+		compatible = "adlink,sema_bmc"; /* no driver needed */
+		reg = <0x28>;
+	};
+
 	ltc1760@0a {			/* Battery charger on LEC-Base R1 carrier */
 		compatible = "linear,ltc1760"; /* guessing, no driver found */
 		reg = <0x0a>;
-- 
2.7.4

